stream_alert/rule_processor/handler.py

```diff
@@ -167,68 +171,141 @@ def get_alerts(self):
         """
         return self._alerts
 
+    @staticmethod
+    def _segment_records_by_count(record_list, max_count):
+        """Segment records by length
+
+        Args:
+            record_list (list): The original records list to be segmented
+            max_count (int): The max amount of records to yield per group
+        """
+        for item in range(0, len(record_list), max_count):
+            yield record_list[item:item + max_count]
+
+    def _segment_records_by_size(self, record_batch):
+        """Segment record groups by size
+
+        Args:
+            record_batch (list): The original record batch to measure and segment
+
+        Returns:
+            generator: Used to iterate on each newly segmented group
+        """
+        split_factor = 1
+        len_batch = len(record_batch)
+
+        while True:
+            if len(str(record_batch[:len_batch / split_factor])) > MAX_BATCH_SIZE:
+                split_factor += 1
+            else:
+                break
+
+        return self._segment_records_by_count(record_batch, len_batch / split_factor)
+
+    @staticmethod
+    def _limit_record_size(batch):
+        """Limit the record size to be sent to Firehose
+
+        Args:
+            batch (list): Record batch to iterate on
+        """
+        for index, record in enumerate(batch):
+            if len(str(record)) > MAX_RECORD_SIZE:
+                # Show the first 1k bytes in order to not overload
+                # CloudWatch logs
+                LOGGER.error('The following record is too large'
+                             'be sent to Firehose: %s', str(record)[:1000])
+                MetricLogger.log_metric(FUNCTION_NAME,
+                                        MetricLogger.FIREHOSE_FAILED_RECORDS,
+                                        1)
+                batch.pop(index)
+
+    def _sanitize_keys(self, record):
+        """Remove special characters from parsed record keys
+
+        This is required when searching in Athena.  Keys can only have
+        a period or underscore
+
+        Args:
+            record (dict): Original parsed record
+
+        Returns:
+            dict: A sanitized record
+        """
+        new_record = {}
+        for key, value in record.iteritems():
+            # Set a default value of the original key
+            sanitized_key = key
+            if re.search(self.special_char_regex, key):
+                sanitized_key = re.sub(self.special_char_regex,
+                                       self.special_char_sub,
+                                       key)
```

Comment:
You don't actually have to check for existence with `re.search`; `re.sub` will leave the original unchanged if there were no matches. So lines 238-242 can be simply:

```python
sanitized_key = re.sub(self.special_char_regex, self.special_char_sub, key)
```